src/main/java/com/google/sps/servlets/AggregationServlet.java

```diff
@@ -50,25 +40,31 @@ public void doGet(HttpServletRequest request, HttpServletResponse response) thro
       response.getWriter().println(e.getMessage());
       return;
     }
-
-    List<ChromeOSDevice> devices = amassDevices();
-    MultiKeyMap<String, Integer> data = processData(devices, fields);
-
-    response.setStatus(HttpServletResponse.SC_OK);
-    response.getWriter().println(Json.toJson(new AggregationResponse(data, fields)));
+    try {
+      List<ChromeOSDevice> devices = amassDevices();
+      MultiKeyMap<String, Integer> data = processData(devices, fields);
+      response.setStatus(HttpServletResponse.SC_OK);
+      response.getWriter().println(Json.toJson(new AggregationResponse(data, fields)));
+    } catch (IOException e) {
+      response.sendRedirect("/login");
+      return;
+    } catch (Exception e) {
+      response.setStatus(HttpServletResponse.SC_BAD_REQUEST);
```

Comment:
This should be SC_INTERNAL_SERVER_ERROR (500) instead? 

5xx are server errors that mean there's something wrong from our side. 4xx (like the SC_BAD_REQUEST) are client errors that mean the user sent something wrong to us.